net/openvswitch/files/openvswitch.init

```diff
@@ -95,6 +95,29 @@ ovs_xx() {
 	esac
 }
 
+ovs_bridge_parse_datapath_id() {
+	local dpid dpif dpnet
+
+	case "$1" in
+		@*)
+			dpnet="${1##@}"
+			dpif="$(uci -q get network."${dpnet}".device)"
+			dpid="6f77$(cat /sys/class/net/${dpif}/address 2>/dev/null | sed 's/://g')"
```

Comment:
Why `6f77`? I am being curious ;)

Reply:
It's explained at the end of the commit message - hex for "ow".

Reply:
Maybe better to change it to 4f 57 (OW), as we write OpenWrt ;-)

Reply:
Or use mac address as is by prefixing with `0x`.  All okay with me.

> The value must either have a 0x prefix or be exactly 16 hex digits long.

The other thing is that maybe we could directly specify linux net device name after `@`, instead of netifd interface name.  It's more straightforward.

Reply:
Didn't know that value could have a 0x prefix and not be 16 digits long. I'll have to think about this again, if this is the case, we might report a valid value as invalid.

As for the linux net device name after @, I used UCI interface config section name, as we use this already in netifd (e.g. for wan6 we use @wan by default). Or did this change in the recent netifd changes?

Reply:
> As for the linux net device name after @, I used UCI interface config section name, as we use this already in netifd (e.g. for wan6 we use @wan by default). Or did this change in the recent netifd changes?

Likely `@wan` works just like before.  It's just that if we want value from a linux net device, use it directly.  No need for a redirection though network uci config.

It may worth mentioning that mac address of ovs bridge itself may change before and after the first netdevice with non-local mac address is added as port.  In this case mac address of the bridge can be somewhat "volatile".  On the other hand, member net devices of a bridge usually do not appear as value of `device` option of a netifd network interface section.

Reply:
In that case I'd rather leave this part out completely, and only offer the user the possibility to enter a valid datapath ID. This will avoid confusing by @ doing one thing here, and another thing in /etc/config/network. This also saves us the trouble of having to warn the user about using interfaces with possibly volatile MAC addresses. The whole idea of adding this feature is to avoid the datapath ID changing at all. So maybe let's keep it simple. What do you think?

Reply:
Yeah, start small and add when needed and ready